cfme/tests/cli/test_appliance_console_db_restore.py

```diff
@@ -0,0 +1,498 @@
+    appl2.ssh_client.put_file(dump_filename, "/tmp/evm_db.backup")
+
+
+def setup_nfs_samba_backup(appl1):
+    # Fetch db from first appliance and push it to nfs/samba server
+    connect_kwargs = {
+        'hostname': cfme_data['network_share']['hostname'],
+        'username': credentials['ssh']['username'],
+        'password': credentials['depot_credentials']['password']
+    }
+    loc = cfme_data['network_share']['nfs_path']
+    nfs_smb = SSHClient(**connect_kwargs)
+    dump_filename = "/tmp/db_dump_{}".format(fauxfactory.gen_alphanumeric())
+    appl1.ssh_client.get_file("/tmp/evm_db.backup", dump_filename)
+    nfs_smb.put_file(dump_filename, "{}share.backup".format(loc))
+
+
+@pytest.mark.tier(2)
+@pytest.mark.uncollectif(lambda appliance: not appliance.is_downstream)
+def test_appliance_console_restore_db_local(request, get_appliances_with_providers):
+    """ Test single appliance backup and restore, configures appliance with providers,
+    backs up database, restores it to fresh appliance and checks for matching providers.
+    """
+    appl1, appl2 = get_appliances_with_providers
+    # Transfer v2_key and db backup from first appliance to second appliance
+    fetch_v2key(appl1, appl2)
+    fetch_db_local(appl1, appl2)
+    # Restore DB on the second appliance
+    appl2.evmserverd.stop()
+    appl2.db.drop()
+    appl2.db.create()
+    command_set = ('ap', '', '4', '1', '', TimedCommand('y', 60), '')
+    appl2.appliance_console.run_commands(command_set)
+    appl2.start_evm_service()
+    appl2.wait_for_web_ui()
+    # Assert providers on the second appliance
+    assert set(appl2.managed_provider_names) == set(appl1.managed_provider_names), (
+        'Restored DB is missing some providers'
+    )
+    # Verify that existing provider can detect new VMs on the second appliance
+    virtual_crud = provider_app_crud(VMwareProvider, appl2)
+    vm = provision_vm(request, virtual_crud)
+    assert vm.provider.mgmt.is_vm_running(vm.name), "vm running"
+
+
+@pytest.mark.tier(2)
+@pytest.mark.uncollectif(lambda appliance: not appliance.is_downstream or appliance.version < '5.9')
+def test_appliance_console_restore_pg_basebackup_ansible(get_appliance_with_ansible):
+    appl1 = get_appliance_with_ansible
+    # Restore DB on the second appliance
+    appl1.evmserverd.stop()
+    appl1.db.restart_db_service()
+    command_set = ('ap', '', '4', '1', '/tmp/backup/base.tar.gz', TimedCommand('y', 60), '')
+    appl1.appliance_console.run_commands(command_set)
+    appl1.start_evm_service()
+    appl1.wait_for_web_ui()
+    appl1.reboot()
+    appl1.ssh_client.run_command(
+        'curl -kL https://localhost/ansibleapi | grep "Ansible Tower REST API"')
+    repositories = appl1.collections.ansible_repositories
+    repository = repositories.create('example', REPOSITORIES[0], description='example')
+    view = navigate_to(repository, "Details")
+    refresh = view.toolbar.refresh.click
+    wait_for(
```

Comment:
Required: Please add message for wait_for
ex. https://github.com/ManageIQ/integration_tests/blob/2f3375e21cec4041e42852e1f4e0673ef11b49db/cfme/tests/control/__init__.py#L43